client/src/context/DataContext.jsx

```diff
@@ -1,16 +1,23 @@
 import React, { createContext, useContext, useState, useEffect } from "react";
+import { usePage } from "./PageContext";
 
 const DataContext = createContext();
 
 export const DataProvider = ({ children }) => {
   const [data, setData] = useState([]);
+  const { currentPage } = usePage();
 
   const baseApiUrl = process.env.REACT_APP_API_URL;
+  let endpoint = "/api/get-workouts";
```

Comment:
let endpoint = currentPage === "profile" ? "/api/get-workouts-for-user" : "/api/get-workouts"

could be a one-liner, and hypothetically would be nicer to be cleaned-up possibly with an enum (a symbolic name for a set of values) to store strings like API endpoints. This way, in the future if we need to change the strings of API endpoints then we can just change from one source to update them all.

just offering an alternative!

Reply:
// ENUM.js

const API = {
Workouts: {
  GET: "/api/get-workouts",
  POST: "/api/post-workouts"
}
}

etc